### PR TITLE
chore: fix cspell errors and enable spell-check in planning directories

### PR DESCRIPTION
**Pull request by HansRobo**
## Description

- fix all cspell errors in planning directories
- enable `spell-check-partial` for planning directories 

## Tests performed

- build test by GithHub Actions
- `spell-check-partial` by GitHub Actions

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by takayuki5168 — APPROVED** [review]

File: planning/behavior_velocity_intersection_module/src/util.cpp [hunk 1]
```diff
@@ -778,15 +778,15 @@ std::vector<DescritizedLane> generateDetectionLaneDivisions(
     auto & branch = branches[(ind2id[src])];
     int node_iter = ind2id[src];
     while (true) {
-      const auto & dsts = adjacency[(id2ind[node_iter])];
+      const auto & distances = adjacency[(id2ind[node_iter])];
```
Comment (soblin) [inline planning/behavior_velocity_intersection_module/src/util.cpp:781]: @HansRobo I was meaning to abbreviate `destination` by `dst` in contrast to `source` and `src`. Can you substitute `destinations` instead ? Thanks

**Review by soblin — CHANGES_REQUESTED** [review]

File: planning/behavior_velocity_intersection_module/src/util.cpp (hunk 1 above)
Comment (HansRobo) [reply planning/behavior_velocity_intersection_module/src/util.cpp:781]: Thanks! I'll fix it

Comment (HansRobo) [reply planning/behavior_velocity_intersection_module/src/util.cpp]: fixed in https://github.com/autowarefoundation/autoware.universe/pull/4716/commits/c059813b14702c5bc82300a4e31a96027443a525

**Review by soblin — APPROVED** [review]

**Comment by HansRobo** [comment]
@TakaHoribe @rej55 @mkuri 
Could you review as code owners of `motion_velocity_smoother`?

**Review by rej55 — APPROVED** [review]

**Comment by rej55** [comment]
@HansRobo Sorry for late review. Thank you for fix!

**Pull request merged**